app/assets/javascripts/discourse/app/services/route-history.js

```diff
@@ -0,0 +1,36 @@
+import { tracked } from "@glimmer/tracking";
+import Service, { service } from "@ember/service";
+import { bind } from "discourse/lib/decorators";
+import { defaultHomepage } from "discourse/lib/utilities";
+
+// This service is responsible for managing the route history
+// mainly used by the `BackButton` component
+export default class RouteHistory extends Service {
+  @service router;
+
+  @tracked lastUrl = null;
+
+  init() {
+    super.init(...arguments);
+    this.router.on("routeWillChange", this.routeWillChange);
+  }
+
+  willDestroy() {
+    this.router.off("routeWillChange", this.routeWillChange);
+  }
+
+  @bind
+  routeWillChange() {
+    this.lastUrl = this.router.currentURL;
+  }
```

Comment:
Surely this is also going to add admin URLs to the history?

So then the "back to forum" button will just be a simple "back" button... and will just take you to the previous admin page you were viewing?

Reply:
> Surely this is also going to add admin URLs to the history

Yes, it did... 😢 

Doing only on component constructor could solve _some_ cases but there are some(refreshing in the admin page) that could make the route history tracking a bit hard
```js
 lastForumUrl = this.routeHistory.lastKnownURL;
```